pandas/_typing.py

```diff
@@ -41,7 +42,8 @@
 
 Dtype = Union[str, np.dtype, "ExtensionDtype"]
 FilePathOrBuffer = Union[str, Path, IO[AnyStr]]
-FrameOrSeries = TypeVar("FrameOrSeries", bound="NDFrame")
+FrameOrSeries = Union["DataFrame", "Series"]
+FrameOrSeriesT = TypeVar("FrameOrSeriesT", bound="NDFrame")
```

Comment:
can you add a comment explaining the difference between FrameOrSeries vs FrameOrSeriesT

Reply:
Hmm after viewing the diff here I am second guess this a bit - is this really just for one-off things like concat? If so probably worth just dealing with exceptions in the individual signatures

Reply:
I don't have a strong opinion on this myself. I can remove it here and if we want this union in _typing.py later, it can be added then?

Reply:
I would be more descriptive on the name and add a comment before these

Reply:
Maybe ``SameFrameOrSeries``? It's difficult to find a descriptive name for the TypeVar that isn't very long. It will be used in a lot of places, so shouldn't be too long IMO.

Reply:
sure that sounds better, definitely add some comments / sections here in typing with liberal explanations on when to use these

Reply:
I find `SameFrameOrSeries` is ambiguous - is that supposed to represent the type or instance of the object being the same? Should be type as far as annotations are concerned but I don't think that's clear

I would be +1 with @simonjayhawkins naming suggestion as I think the `T` is a pseudo-convention seen most with TypeVars in typeshed